addons/web/static/src/js/views/basic/basic_controller.js

```diff
@@ -263,31 +263,19 @@ var BasicController = AbstractController.extend(FieldManagerMixin, {
      * @returns {Promise}
      */
     _callButtonAction: function (attrs, record) {
-        var self = this;
-        var def = new Promise(function (resolve, reject) {
-            var reload = function () {
-                return self.isDestroyed() ? Promise.resolve() : self.reload();
-            };
-            record = record || self.model.get(self.handle);
-
-            self.trigger_up('execute_action', {
-                action_data: _.extend({}, attrs, {
-                    context: record.getContext({additionalContext: attrs.context || {}}),
-                }),
-                env: {
-                    context: record.getContext(),
-                    currentID: record.data.id,
-                    model: record.model,
-                    resIDs: record.res_ids,
-                },
-                on_success: resolve,
-                on_fail: function () {
-                    self.update({}, { reload: false }).then(reject).guardedCatch(reject);
-                },
-                on_closed: reload,
-            });
-        });
-        return this.alive(def);
+        record = record || this.model.get(this.handle);
+        const actionData = Object.assign(
+            {},
+            attrs,
+            { context: record.getContext({additionalContext: attrs.context || {}}) }
+        );
```

Comment:
```js
const actionData = Object.assign(
    {},
    attrs,
    { context: record.getContext({additionalContext: attrs.context || {}}) }
);
```

```js
const actionData = Object.assign({}, attrs, {
    context: record.getContext({ additionalContext: attrs.context || {} }),
});
```

I personally prefer the second one (which is, I think, plus en phase avec nos conventions)